testing/asserts.ts

```diff
@@ -641,66 +641,87 @@ export function assertIsError<E extends Error = Error>(
  * callback which will be passed the error, usually to apply some custom
  * assertions on it.
  */
+export function assertThrows(
+  fn: () => unknown,
+  msg?: string,
+): unknown;
 export function assertThrows<E extends Error = Error>(
   fn: () => unknown,
   // deno-lint-ignore no-explicit-any
-  ErrorClass?: new (...args: any[]) => E,
+  ErrorClass: new (...args: any[]) => E,
   msgIncludes?: string,
   msg?: string,
-): void;
+): E;
+export function assertThrows(
+  fn: () => unknown,
+  msgIncludes: string,
+  msg?: string,
+): Error;
```

Comment:
When I run the code like the below, it doesn't throw the assertion error (`msgIncludes` is not compared with thrown error message). I think this signature isn't supported by this change. Why is this signature added? How about just removing this?

```ts
import { assertThrows } from "./testing/asserts.ts";

assertThrows(
  () => {
    throw new Error("foo");
  },
  "bar",
  "baz",
);
```